aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -544,7 +548,7 @@ void fmod_kernel(TensorIterator& iter) {
       });
     });
   } else {
-    AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "fmod_cpu", [&]() {
+    AT_DISPATCH_FLOATING_TYPES_AND(kHalf, iter.dtype(), "fmod_cpu", [&]() {
```

Comment:
Was there any reason not to use the `AT_DISPATCH_FLOATING_TYPES_AND_HALF` macro, see also ln 573?

Reply:
Maybe should become `AT_DISPATCH_FLOATING_TYPES_AND2(kBFloat16, kHalf, iter.dtype(), ... `?

Reply:
I noticed the deprecation message for `AT_DISPATCH_ALL_TYPES_AND_HALF` and figured the `_AND` variants were preferred.
https://github.com/pytorch/pytorch/blob/87e97ced2004e2ea0e97ed380a020c2281e76204/aten/src/ATen/Dispatch.h#L50-L51

Reply:
@peterbell10 sry, I was not aware of the deprecation.